lib/go/Makefile

```diff
@@ -15,15 +15,34 @@ endif
 export GOPATH
 
 
+################################################################################
+##                                   DEP                                      ##
+################################################################################
+
+DEP ?= ./dep
+DEP_VER ?= 0.3.1
+DEP_BIN := dep-$$GOHOSTOS-$$GOHOSTARCH
+DEP_URL := https://github.com/golang/dep/releases/download/v$(DEP_VER)/$$DEP_BIN
+
+$(DEP):
+	GOVERSION=$$(go version | awk '{print $$4}') && \
+	GOHOSTOS=$$(echo $$GOVERSION | awk -F/ '{print $$1}') && \
+	GOHOSTARCH=$$(echo $$GOVERSION | awk -F/ '{print $$2}') && \
+	DEP_BIN="$(DEP_BIN)" && \
+	DEP_URL="$(DEP_URL)" && \
+	curl -sSLO $$DEP_URL && \
+	chmod 0755 "$$DEP_BIN" && \
+	mv -f "$$DEP_BIN" "$@"
+
+vendor: | $(DEP)
```

Comment:
maybe a compromise would be a `vendor: check-dep` dependency, where `check-dep` tests for the presence of `dep` in the $PATH and if it's missing exits the build?

if we want to stay friendly, we could also have a target `dep: | $(DEP)` so that a user that's missing `dep` could simply invoke `make dep` to pull it down.

Reply:
Or we merge as is and handle it later since this adds value without detracting from any. This needs to be deterministic, period. Or bindings could be off since the tool used to grab the eps may influence the eps.